File: bfs.cpp
#include<iostream>
#include<vector>
#define MAX 5

using namespace std;

void bfs(vector<vector<int>>&matrix,int start)
{
	vector<bool>v(MAX, false);
	vector<int>q;
	q.push_back(start);
	
	visited[start] = true;
	int vis;
	while(!q.empty())	
}

int main()
{
	int size = 5;
	vector<vector<int>>matrix;
	matrix = {{0,0,0,1,1},
		{0,0,1,1,1,},
		{0,1,0,1,0},
		{1,1,1,0,1},
		{1,0,1,1,0}};

	bfs(matrix,0);
	return 0;
}

File: bfs.py
from collections import deque

def bfs(matrix, start):
    visited = [False] * len(matrix)
    queue = deque()
    queue.append(start)
    visited[start] = True
    
    while queue:
        node = queue.popleft()
        print(node, end=" ")
        for i in range(len(matrix[node])):
            if matrix[node][i] == 1 and not visited[i]:
                queue.append(i)
                visited[i] = True

if __name__ == "__main__":
    matrix = [
        [0, 0, 0, 1, 1],
        [0, 0, 1, 1, 1],
        [0, 1, 0, 1, 0],
        [1, 1, 1, 0, 1],
        [1, 0, 1, 1, 0]
    ]
    bfs(matrix, 0)
